### DISCUSSION TITLE
CUDAcross platform macOS and linux

### DISCUSSION THREAD

**Discussion by the-code-samurai-97** [bazel-contrib/rules_cuda#201] [Q&A]
What is the best way to not see any errors when i do bazel build //... i am trying platforms but i still loading the rules cuda using https archive so it still give me error
` While resolving toolchains for target //src/examples/cuda_example:kernel: No matching toolchains found for types @rules_cuda//cuda:toolchain_type.
`

**Comment by cloudhan** [comment]
Great question, macOS is not supported, then there should be a way to exclude cuda targets from __all__ when you are building on the platfrom.

One way to add `target_compatible_with = ["@platforms//os:<blahblah>"]` to the target, but it is not scalable. This will be easy if you are developing a very small project with a few cuda targets.

Another way is to leverage [if_cuda](https://github.com/bazel-contrib/rules_cuda/tree/main/examples/if_cuda), it basically conditionally disable all cuda target on unsupported platfroms, but we need to improve it (if possible) so that it automatically excludes from __all__...
